### PR TITLE
CONFIG: Added configuration file with section for Grace Hopper.

### PR DESCRIPTION
**Pull request by rakhmets**
## What
Added configuration file with section for Grace Hopper, setting `REG_NONBLOCK_MEM_TYPES=host,cuda-managed`.

### CODE REVIEW

File: config/ucx.conf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[Grace Hopper]
+Product name=*Grace Hopper*
```
Comment (brminich) [inline config/ucx.conf:2]: maybe this
```suggestion
Product name=*Grace*Hopper*
```

**Review by brminich — DISMISSED** [review]

File: config/Makefile.am [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (C) 2024 NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# See file LICENSE for terms.
+#
+
+conf_files_dir = $(ucx_config_dir)
+conf_files__DATA = ucx.conf
```
Comment (yosefe) [inline config/Makefile.am:8]: i think need single underscore, not a double

File: config/ucx.conf [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[Grace Hopper]
+Product name=*Grace*Hopper*
+UCX_REG_NONBLOCK_MEM_TYPES=host,cuda-managed
```
Comment (yosefe) [inline config/ucx.conf:3]: why cuda-managed?

File: config/Makefile.am (hunk 2 above)
Comment (rakhmets) [reply config/Makefile.am:8]: With single it will be:
```
conf_filesdir = $(ucx_config_dir)
conf_files_DATA = ucx.conf
```

Comment (yosefe) [reply config/Makefile.am:8]: IMO it is better

File: config/ucx.conf (hunk 3 above)
Comment (rakhmets) [reply config/ucx.conf]: Removed.

**Review by yosefe — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Pull request merged**